### PR TITLE
Add pageNumber

### PR DESCRIPTION
**Pull request by pgh70**
The prev and next links in _links describe to provide a link to the next resultset of a list. No parameter has been specified to accomplish this. To standardize this a pageNumber is added.

Also fixes some issue's with the type:number and format:int32 combination. This combination isn't valid according to OpenAPI v3 spec.

### CODE REVIEW

**Comment by joostfarla** [comment]
I now remember why we chose to leave the `pageNumber` parameter out of the specification. Most databases are compatible with a simple page number, because you can calculate an "offset" value (e.g. `pageSize` * (`pageNumber` - 1)) from it. But this is not the case for every database. If you look at DynamoDB (AWS) for example, you have to pass the identifier of the next document (which is a set of 1 or two key-value pairs with strings/numbers) as offset. It cannot be calculated by using the `pageNumber`. Therefore, to not restrict the underlying database type used, I think we should leave it this way. The `next` and `prev` links should suffice to do pagination.

File: paths/BuildingCollection.yaml [hunk 1]
```diff
@@ -14,13 +14,18 @@ get:
             type: object
             required:
               - pageSize
+              - pageNumber
               - _embedded
               - _links
             properties:
               pageSize:
-                type: number
+                type: integer
```
Comment (joostfarla) [inline paths/BuildingCollection.yaml]: Can you fix this for the whole spec in the oas3 branch?

**Comment by joostfarla** [comment]
> based on this reasoning, we have actually chosen to include it our implementation: it would not work without it :) We used it to make the next and prev links to work, as if we would not paginate before doing so, the next and prev links would not work on our side . 

Every institution is free to add implementation-specific parameters to make this work. It's just not part of the OOAPI spec. Clients should blindly follow next/prev links without awareness of the parameters used.

**Comment by pgh70** [comment]
I made the pageNumber completely optional. It can be seen as a suggestion how to implement paging to create next and prev links. Of course every institute is free to implement it another way, but by suggesting a pageNumber we prevent reinvention of the wheel by each institute.

File: paths/BuildingCollection.yaml (hunk 1 above)
Comment (pgh70) [reply paths/BuildingCollection.yaml]: I did in commit: 0c4f6bb

**Comment by fransward** [comment]
Merged to prepare for V3.0 voting

**Pull request merged**